lib/cisco_node_utils/dns_domain.rb

```diff
@@ -59,8 +59,24 @@ def create
         config_set('dnsclient', 'domain_list',
                    state: '', name: @name)
       else
-        config_set('dnsclient', 'domain_list_vrf',
-                   state: '', name: @name, vrf: @vrf)
+        # On some platforms attempts to create a new domain-list results
+        # in the error. 'ERROR: Deletion of VRF test in progresswait
+        # for it to complete'.  We handle this by trying up to 500 times
+        # before giving up.
+        tries = 500
```

Comment:
Looks like you're retrying as fast as possible. Would it make more sense to retry up to a max of, say, 30 times, but sleep 1 second between each try?

How long does 500 retries take? What's the maximum number of retries you've seen before it succeeds?